codegenerator/cli/templates/static/codegen/src/eventFetching/ChainFetcher.res

```diff
@@ -375,3 +375,8 @@ let rollbackToLastBlockHashes = (chainFetcher: t, ~rolledBackLastBlockData) => {
 let isFetchingAtHead = (chainFetcher: t) =>
   chainFetcher.fetchState->PartitionedFetchState.isFetchingAtHead
 
+let getFirstEventBlockNumber = (chainFetcher: t) =>
+  Utils.Math.minOptInt(
+    chainFetcher.dbFirstEventBlockNumber,
+    chainFetcher.fetchState->PartitionedFetchState.getFirstEventBlockNumber,
+  )
```

Comment:
Evaluate the DBs first event block number against the partions first event block number in the case of an indexer restart.